usaspending_api/awards/management/commands/generate_unlinked_awards_download.py

```diff
@@ -0,0 +1,218 @@
+            except Exception:
+                logger.exception("Exception encountered. See logs")
+                raise
+            finally:
+                # "best-effort" attempt to cleanup temp files after a failure. Isn't 100% effective
+                self.cleanup()
+
+    def _create_data_csv_dest_path(self, file_name) -> Path:
+        return self.working_dir_path / file_name
+
+    def process_data_copy_jobs(self, zip_file_path):
+        logger.info(f"Creating new unlinked awards download zip file: {zip_file_path}")
+        self.filepaths_to_delete.append(zip_file_path)
+
+        for delta_table_name, sql_file, final_name in self.download_file_list:
+            df = self.spark.sql(f"select * from {delta_table_name} where toptier_code = '{self._toptier_code}'")
+            sql_file = None
+            final_path = self._create_data_csv_dest_path(final_name)
+            intermediate_data_file_path = final_path.parent / (final_path.name + "_temp")
+            data_file_names, count = self.download_to_csv(
+                sql_file, final_path, final_name, str(intermediate_data_file_path), zip_file_path, df
+            )
+            if count <= 0:
+                logger.warning(f"Empty data file generated: {final_path}!")
+
+            self.filepaths_to_delete.extend(self.working_dir_path.glob(f"{final_path.stem}*"))
+
+    def complete_zip_and_upload(self, zip_file_path):
+        self.finalize_zip_contents(zip_file_path)
+        if self.upload:
+            logger.info("Upload final zip file to S3")
+            upload_download_file_to_s3(zip_file_path, settings.UNLINKED_AWARDS_DOWNLOAD_REDIRECT_DIR)
+            logger.info("Marking zip file for deletion in cleanup")
+        else:
+            logger.warn("Not uploading zip file to S3. Leaving file locally")
+            self.filepaths_to_delete.remove(zip_file_path)
+
+    @property
+    def download_file_list(self):
+        short_timestamp = self.full_timestamp[:-6]
+        return [
+            (
+                "summary_file",
+                f"{self.download_source_sql['summary_file']}",
+                f"{self._agency_name}_UnlinkedAwardsSummary_{short_timestamp}",
+            ),
+            (
+                "unlinked_contracts_file_d1",
+                f"{self.download_source_sql['unlinked_contracts_file_d1']}",
+                f"{self._agency_name}_UnlinkedContracts_{short_timestamp}",
+            ),
+            (
+                "unlinked_assistance_file_d2",
+                f"{self.download_source_sql['unlinked_assistance_file_d2']}",
+                f"{self._agency_name}_UnlinkedFinancialAssistance_{short_timestamp}",
+            ),
+            (
+                "unlinked_accounts_file_c",
+                f"{self.download_source_sql['unlinked_accounts_file_c']}",
+                f"{self._agency_name}_UnlinkedAccountsByAward_{short_timestamp}",
+            ),
+        ]
+
+    def cleanup(self):
```

Comment:
These four methods seem to have a lot in common with their corresponding methods in `generate_covid_19_download.py`. Did you consider any way of making sharing code between commands? Maybe inheriting between a common Base Class or simply converting these to helper functions instead of class methods?

Reply:
No because they only have things in common by coincidence. They very well could deviate from each other.

Reply:
Actually, when I compare the methods, they are in fact not identical. There's a lot of deviations.

Reply:
The covid-19 and unlinked awards commands both cleanup in a different manner. The covid-19 and unlinked awards commands both `process_data_copy_jobs` in a different manner. The covid-19 and unlinked awards commands both `finalize_zip_contents` in a different manner. Additionally, we find different parameters to these methods between  covid-19 and unlinked awards commands.

Reply:
Fair enough, I can see that there are some differences there in those methods.